TAScheduler/UnitTests/tests.py

```diff
@@ -76,71 +77,879 @@ class TestAdminCourseInstrAsgmt(TestCase):  # Kevin
+        self.taObj.assignTACourse(self.courseDB)
+        self.assertEquals(self.taObj.getTACrseAsgmts(), 1, msg="should be 1 assigment")
+
+    # 2] 0 assignment
+    def test_0Assignment(self):
+        self.assertEquals(self.taObj.getTACrseAsgmts(), 0, msg="should be 0 assigments")
+
+    # 3] 1 "assignment" - not in db
+    def test_1AssignmentNoExistCourse(self):
+        tempCourse = Course(course_id=102)  # not "101", which exists already
+        self.taObj.assignTACourse(tempCourse)
+        self.assertEquals(self.taObj.getTACrseAsgmts(), 0, msg="shouldn't assign non-existing course")
+
+    # 4] 1->0 Assignment
+    def test_1to0Assignment(self):
+        self.taObj.assignTACourse(self.courseDB)
+        self.adminObj.removeCourse(self.courseDB)
+        self.assertEquals(self.taObj.getTACrseAsgmts(), 0, msg="added then removed course, should be 0")
 
 
 class TestAssignTALab(TestCase):
-    pass
+    taDB = None
+    courseDBList = list()  # just for section
+    sectionDBList = list()  # just for lab
+    labDBList = list()
+    user = None  # for TA
+    taObj = None
+
+    def setUp(self):
+        self.user = User.objects.create(
+            email_address='TA@example.com',
+            password='TApassword',
+            first_name='TA',
+            last_name='User',
+            home_address='123 TA St',
+            phone_number='1234567890'
+        )
+        self.taDB = TA.objects.create(user=self.user, max_assignments=2)  # max 2 assignment!
+        self.taObj = TAObj(self.taDB)  # creating TA object using TA in database
+
+        for i in [1, 2, 3]:  # courses
+            self.courseDBList.append(Course.objects.create(
+                course_id=100 + i,
+                semester='Fall 2023',
+                name='Introduction to Testing',
+                description='A course about writing tests in Django.',
+                num_of_sections=3,
+                modality='Online',
+                credits=4
+            ))
+        for i in [1, 2, 3]:  # section
+            self.sectionDBList.append(Section.objects.create(
+                section_id=100 + i,
+                course=self.courseDBList[i],
+                location="location" + str(i),
+                meeting_time="mt" + str(i)
+            ))
+        for i in [1, 2, 3]:  # lab
+            self.labDBList.append(Lab.objects.create(
+                section_id=self.sectionDBList[i],
+                ta=None  # HOPEFULLY FINE?
+            ))
+
+    # 1] Success TA->Lab
+    def test_ExistLab(self):
+        self.taObj.assignTALab(self.labDBList[0])
+        self.assertIn(self.taObj, Lab.objects.get(ta=self.taDB).ta
+                      , "Should have linked lab and TA together")
+
+    # 2] Adding Lab not existing in DB
+    def test_NotExistLab(self):
+        tempLab = Lab(section=self.sectionDBList[0])  # created, not saved to db
+        with self.assertRaises(ValueError, msg="can't send in non existing lab, i.e., lab obj"):
```

Comment:
TypeError not value error

Reply:
Also this is the same test as test_Nonlab

Reply:
ValueError: "Python's ValueError occurs when a function is called with the proper argument type but with the wrong value."  This test send in a tempLab which is the correct argument type (a lab obj) but it's value is incorrect.